packages/core/react/use-chat.ts

```diff
@@ -159,10 +159,21 @@ const getStreamedResponse = async (
 
   if (isComplexMode) {
     while (true) {
-      const { done, value } = await reader.read()
+      const { done, value: partialValue } = await reader.read()
       if (done) {
         break
       }
+
+      let value = partialValue
+      // while last character of value is not a newline, keep reading
+      while (value[value.length - 1] !== 10) {
+        const { done, value: partialValue } = await reader.read()
+        if (done) {
+          break
+        }
+        value = new Uint8Array([...value, ...partialValue])
```

Comment:
I think we can do better than constantly re-allocating a new array but this works for now

Reply:
I have updated the diff: now a buffer is used and is reallocated exponentially whenever there is not enough space. :)